src/plugins/intel_npu/src/plugin/src/model_version.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "model_version.hpp"
+
+#include <cstring>
+#include <sstream>
+
+#include "intel_npu/utils/logger/logger.hpp"
+#include "openvino/core/version.hpp"
+
+namespace intel_npu {
+
+OpenvinoVersion::OpenvinoVersion(const std::string& version) {
+    this->version = version;
+    this->size = static_cast<uint32_t>(version.size());
+}
+
+void OpenvinoVersion::read(std::istream& stream) {
+    stream.read(reinterpret_cast<char*>(&size), sizeof(size));
+    stream.read(&version[0], size);
+}
+
+Metadata<1, 0>::Metadata() : version{1, 0}, ovVersion{ov::get_openvino_version().buildNumber} {}
+
+void Metadata<1, 0>::read(std::istream& stream) {
+    ovVersion.read(stream);
+}
+
+void Metadata<1, 0>::write(std::ostream& stream) {
+    stream.write(reinterpret_cast<const char*>(&version.major), sizeof(version.major));
+    stream.write(reinterpret_cast<const char*>(&version.minor), sizeof(version.minor));
+
+    stream.write(reinterpret_cast<const char*>(&ovVersion.size), sizeof(ovVersion.size));
+    stream.write(ovVersion.version.c_str(), ovVersion.version.size());
+}
+
+std::unique_ptr<MetadataBase> createMetadata(int major, int minor) {
+    switch (major) {
+    case 1:
+        switch (minor) {
+        case 0:
+            return std::make_unique<Metadata<1, 0>>();
+
+        default:
+            return nullptr;
+        }
+
+    default:
+        return nullptr;
+    }
+}
+
+bool Metadata<1, 0>::isCompatible() {
+    // checking if we still support the format
+    // but is checking `Major` redundant since it's checked in createMetadata?
+    if (version.major != CURRENT_METAVERSION_MAJOR || version.minor != CURRENT_METAVERSION_MINOR) {
+        return false;
+    }
+    // Checking if we can import the blob
+    return ovVersion.version == ov::get_openvino_version().buildNumber;
```

Comment:
Please help me understand why we need another check if the versions already match.

Reply:
Because we would not accept blobs with different OV versions, whatever `Major.Minor` they got.

And.. if `Major` and `Minor` match, we could at least print info about the blob Metadata contents before we halt the program.